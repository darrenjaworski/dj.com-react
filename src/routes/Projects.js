import React from "react";

import projectData from "../data/projects.json";
import { Heading, SmallHeading } from "../components/heading";
import { TextLink } from "../components/links";
import Paragraph from "../components/paragraph";

const Projects = () => {
  let projects = projectData.map(d => {
    return (
      <React.Fragment key={d.title.replace(" ", "-")}>
        <SmallHeading>
          <TextLink href={d.link}>{d.title}</TextLink>
        </SmallHeading>
        <Paragraph>{d.paragraph}</Paragraph>
      </React.Fragment>
    );
  });
  return (
    <>
      <Heading>projects</Heading>
      {projects}
    </>
  );
};

export default Projects;
