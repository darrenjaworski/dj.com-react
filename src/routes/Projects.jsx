import React from "react";
import { Link } from "react-router-dom";
import projectData from "../data/projects.json";
import { Heading, SmallHeading } from "../components/heading";
import { TextLink } from "../components/links";
import Paragraph from "../components/paragraph";
import styled from "@emotion/styled";

const StyledLink = styled(Link)`
  color: ${(props) => props.theme.linkColor};
  transition: color 0.5s;
  text-decoration: underline;
  font-size: 1.25rem;
`;

const Projects = () => {
  const projects = projectData.map((d) => {
    let link;

    if (d.route) {
      link = (
        <>
          <TextLink href={d.link}>{d.title}</TextLink>
          {" - "}
          <StyledLink to="/frogs">demo</StyledLink>
        </>
      );
    } else {
      link = <TextLink href={d.link}>{d.title}</TextLink>;
    }

    return (
      <React.Fragment key={d.title.replace(" ", "-")}>
        <SmallHeading>{link}</SmallHeading>
        <Paragraph>{d.paragraph}</Paragraph>
      </React.Fragment>
    );
  });
  return (
    <>
      <Heading>projects</Heading>
      {projects}
    </>
  );
};

export default Projects;
