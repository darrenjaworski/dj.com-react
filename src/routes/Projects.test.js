import React from "react";
import renderer from "react-test-renderer";

import Projects from "./Projects";

describe("FourOhFour", () => {
  it("renders without crashing", () => {
    const tree = renderer.create(<Projects />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
