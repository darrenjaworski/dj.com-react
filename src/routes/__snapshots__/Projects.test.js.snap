// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FourOhFour renders without crashing 1`] = `
Array [
  <h1
    className="css-lozscu"
  >
    projects
  </h1>,
  <h1
    className="css-wqkrd8"
  >
    <a
      className="css-alkraz"
      href="https://github.com/darrenjaworski/byte-converter-react"
    >
      byte-converter-react
    </a>
  </h1>,
  <p
    className="css-z4ydkm"
  >
    Small React component to convert bytes.
  </p>,
  <h1
    className="css-wqkrd8"
  >
    <a
      className="css-alkraz"
      href="https://github.com/darrenjaworski/d3-reusable"
    >
      D3re
    </a>
  </h1>,
  <p
    className="css-z4ydkm"
  >
    A small D3 reusable charts library. Basically I'm working to add some missing .chart() functions in the D3 library.
  </p>,
  <h1
    className="css-wqkrd8"
  >
    <a
      className="css-alkraz"
      href="https://github.com/darrenjaworski/js-algorithms/"
    >
      JS algorithms
    </a>
  </h1>,
  <p
    className="css-z4ydkm"
  >
    Working through Algorithms, 4th edition and Introduction to Algorithms, 3rd edition using Javascript and teaching myself all the CS I don't work on in my day job. I'm thinking of expanding this and recreating the data structures and algorithms in Java and Python as well.
  </p>,
  <h1
    className="css-wqkrd8"
  >
    <a
      className="css-alkraz"
      href="https://docs.google.com/presentation/d/1bOAEXV-vtFnRP4jIZYFUXn1QA5f3u0QDjdme_s-NmEQ/edit?usp=sharing"
    >
      Code for OKC D3 Talk
    </a>
  </h1>,
  <p
    className="css-z4ydkm"
  >
    Short talk on D3.
  </p>,
  <h1
    className="css-wqkrd8"
  >
    <a
      className="css-alkraz"
      href="https://github.com/darrenjaworski/coffee-lovers-okc"
    >
      OKC coffee
    </a>
  </h1>,
  <p
    className="css-z4ydkm"
  >
    An app to explore coffee shops in Oklahoma City. Designed and built from the ground up to be a hub for coffee people of OKC. However, a lot more work needs to be done on this project. I need to finish collecting data for coffee shops, to develop a suggestion system to lead you to the perfect cup in town, and to actually launch the thing. I'll take a couple of weekends and finish this project over a cup of coffee.
  </p>,
  <h1
    className="css-wqkrd8"
  >
    <a
      className="css-alkraz"
      href="http://darrenjaworski.github.io/simplestats/"
    >
      stats made simple
    </a>
  </h1>,
  <p
    className="css-z4ydkm"
  >
    A series of explainers and tutorials for aspiring statisticians. This is another project that has significantly more work to be done. I'll need two cups of coffee.
  </p>,
  <h1
    className="css-wqkrd8"
  >
    <a
      className="css-alkraz"
      href="http://bl.ocks.org/darrenjaworski/"
    >
      earthquakes in Oklahoma
    </a>
  </h1>,
  <p
    className="css-z4ydkm"
  >
    I felt an earthquake during lunch. I decided to put a couple of minutes into a series of maps and charts that touch on the swarm of induced earthquakes in Oklahoma. Each visualization is updated with real-time earthquake data. I will include more data layers as they become available.
  </p>,
  <h1
    className="css-wqkrd8"
  >
    <a
      className="css-alkraz"
      href="http://bl.ocks.org/darrenjaworski/f2540e53c4e327a145d0"
    >
      legislature voting trends
    </a>
  </h1>,
  <p
    className="css-z4ydkm"
  >
    Similarity matrix of the 2015 Oklahoma Legislature voting trends. This small project basically answers the question: which legislators vote like each other?
  </p>,
]
`;
